spec/Spec/BlsEC/G2.cry

```diff
@@ -0,0 +1,543 @@
+    }
+
+/**
+ * G2.xCoeffs cubed. Computed using py_ecc.
+ */
+G2_xCubed : Fq2
+G2_xCubed =
+    { real = 3341065098200961989598748404381324054605449840948293400785922068969583005812936621662354076014412578129291257715484
+    , imag = 2133050398774337206222816300118221327418763981033055222570091459262312519047975404484651902003138703421962555090218
+    }
+
+/**
+ * G2.xCoeffs - G2.yCoeffs. Computed using py_ecc.
+ */
+G2_y_minus_x : Fq2
+G2_y_minus_x =
+    { real = 1632449532699825317380915405166111493410839025422880490639470020826550940638108691550521503919468808688732414044745
+    , imag = 1870818876469786139193731976758160400328152687915953296697712908626373202724554996990127355752626211775441853413611
+    }
+
+/**
+ * G2.xCoeffs + G2.yCoeffs. Computed using py_ecc.
+ */
+G2_y_plus_x : Fq2
+G2_y_plus_x =
+    { real = 2337851671874758553755193637188231783190898931008430971079287309306138512109538745297216704607199160594256266651065
+    , imag = 3986698009736546165718461780529819517744210518930240528684401185085612201771633997414458852832304583413319107585340
+    }
+
+/**
+ * G2.yCoeffs / G2.xCoeffs. Computed using py_ecc.
+ */
+G2_y_div_mod_x : Fq2
+G2_y_div_mod_x =
+    { real = 2645964961145567287646174681515652840605182444869364111679220104429646387260976011822898928226591963576020240426619
+    , imag = 3656195751079888906744826605664754595376734064625263022628224160781596224005157110042338963344577397675474692059176
+    }
+
+/**
+ * G2.xCoeffs inverse. Computed using py_ecc.
+ */
+G2_xInverse : Fq2
+G2_xInverse =
+    { real = 1223514866584400226619392018844575446040489926980124362465220784014275923982521236729115905530146470033177964357563
+    , imag = 3495453419063574139925455551631318136524378768480668459768684498547208492074727172949503320416899422709399085629096
+    }
+
+/**
+ * G2.yCoeffs inverse. Computed using py_ecc.
+ */
+G2_yInverse : Fq2
+G2_yInverse =
+    { real = 3828301281232776537393462114696026453820925117135598932697882187079936548530569912296382199133331215710911400309693
+    , imag = 1450185639544495128598198914056492809111071767684112182252696543466687995844478949870392653614483391964324409513904
+    }
+
+/**
+ * G2 negated. Computed using py_ecc: `normalize(neg(G2))`.
+ */
+G2_Negated : G2Point
+G2_Negated =
+    { xCoeffs =
+        { real = 352701069587466618187139116011060144890029952792775240219908644239793785735715026873347600343865175952761926303160
+        , imag = 3059144344244213709971259814753781636986470325476647558659373206291635324768958432433509563104347017837885763365758
+        }
+    , yCoeffs =
+        { real = 2017258952934375457849735304558732518256013841723352154472679471057686924117014146018818524865681679396399932211882
+        , imag = 3074855889729334937670587859959866275799142626485414915307030157330054773488162299461738339401058098462460928340205
+        }
+    }
+
+/*
+ * ==========
+ * Properties
```

Comment:
These are nice... and we should highlight the strength of these proofs

Reply:
My plan is to create a report along with our delivery of the Cryptol code. I'll do that there.